### ISSUE TITLE
Server allows user profile images of ridiculous size

### ISSUE THREAD

**Issue by Daenyth** [Cockatrice/Cockatrice#750]
Look at Ninja Ferret; black image in 7136x2992 resolution

**Comment by ZeldaZach** [comment]
We can add a restriction to the uploader on the webUI service to, say, 300x300?

**Comment by Daenyth** [comment]
I'd like to enforce something in the main code. Probably when we allow
profile modification in client

**Comment by woogerboy21** [comment]
@ctrlaltca Was this worked in already?

**Comment by ctrlaltca** [comment]
Not really; in-client profile editing forces the image to a max 400x200 jpeg. The problem is also that the server's database column is a BLOB, limited to 64kb.

**Comment by ZeldaZach** [comment]
If it's forced to a maximum size, then it seems like this is kinda-fixed

**Comment by tooomm** [comment]
So if somebody uploads a picture within the client it will get resized to 400x200?

I just tried it myself and it looks like the pictures are getting shrinked + they keep the original image ratio.
Do Cockatrice compress them as well to match the 64kb limit?

**Comment by ctrlaltca** [comment]
> So if somebody uploads a picture within the client it will get resized to 400x200?

It will be shrunk (respecting keeping its original width/height ratio) to fit inside that size.

> Does Cockatrice compress them before uploading as well to match the 64kb limit?

No, the image is not compressed before uploading; the raw image ends up in the database, and if it's more that 64kb, it will just be badly truncated by the database. This is indeed a problem that needs to be fixed.

**Comment by tooomm** [comment]
So the issue still standing here is:
>the raw image ends up in the database, and if it's more that 64kb, it will just be badly truncated by the database. This is indeed a problem that needs to be fixed.

Also, there is the different handling for image uploading in-client and the online form as decribed in https://github.com/Cockatrice/Cockatrice/issues/750#issuecomment-181257858

**Comment by D0nV170** [comment]
So issues still persist with images. I was able to get a user to upload ~100kb image with no issues but when she tried to upload a larger image size, when she logged off it greyed out like the last 10% of the image. I saw the issue again on another user and asked how large his avatar image was and he told me ~500kb and almost all of it was greyed out. This seems only to happen at logoff and the image is fixed the next time it's uploaded.

**Comment by ZeldaZach** [comment]
Temp fix was to change field from BLOB to MEDIUMBLOB. I'd still like to add reduction code to the service so it fits into a BLOB

**Comment by ZeldaZach** [comment]
This is related to #793
